test/raw_audio_parser_test.exs

```diff
@@ -0,0 +1,91 @@
+defmodule RawAudioParserTest do
+  @moduledoc false
+
+  use ExUnit.Case, async: true
+  use Membrane.Pipeline
+
+  import Membrane.Testing.Assertions
+
+  alias Membrane.AudioMixer.Support.RawAudioParser
+  alias Membrane.Testing.{Pipeline, Sink, Source}
+  alias Membrane.{Buffer, RawAudio, Time}
+
+  @stream_format %RawAudio{
+    channels: 2,
+    sample_rate: 44_100,
+    sample_format: :s24le
+  }
+
+  @silence RawAudio.silence(@stream_format, Time.milliseconds(10))
+
+  test "parser devides payloads into samples" do
+    payload_bytes = div(byte_size(@silence), 2)
+    <<payload::binary-size(payload_bytes), _rest::binary>> = @silence
+
+    buffers = Enum.map(1..9, fn _idx -> payload end)
+
+    structure = [
+      child(:source, %Source{output: buffers, stream_format: @stream_format})
+      |> child(:parser, RawAudioParser)
+      |> child(:sink, Sink)
+    ]
+
+    assert pipeline = Pipeline.start_link_supervised!(structure: structure)
+
+    extended_payload = RawAudio.silence(@stream_format, Time.milliseconds(5))
+
+    <<^extended_payload::binary-size(byte_size(extended_payload)), truncated_payload::binary>> =
+      @silence
+
+    Enum.each(1..5, fn _idx ->
+      assert_sink_buffer(pipeline, :sink, %Buffer{pts: nil, payload: ^truncated_payload})
+    end)
+
+    Enum.each(1..4, fn _idx ->
+      assert_sink_buffer(pipeline, :sink, %Buffer{pts: nil, payload: ^extended_payload})
+    end)
+
+    assert_sink_buffer(pipeline, :sink, %Buffer{pts: nil, payload: <<0, 0, 0>>})
```

Comment:
Any comment on what's happening here would be appreciated. In particular, what's the difference between extended_payload and truncated_payload. Aren't they always the same?

I also assume, there is some correlation between `Time.milliseconds(5)` in line 35 and silence duration from line 19. If that's correct, then we should calculate `Time.milliseconds(5)` instead of hardcoding it.